include/seqan3/core/algorithm/pipeable_config_element.hpp

```diff
@@ -0,0 +1,150 @@
+    friend constexpr auto operator|(pipeable_config_element && lhs,
+                                    pipeable_config_element<other_derived_t, other_value_t> && rhs)
+    {
+        return configuration{static_cast<derived_t &&>(lhs)}.push_back(static_cast<other_derived_t &&>(rhs));
+    }
+
+    //!\overload
+    template <typename other_derived_t, typename other_value_t>
+    friend constexpr auto operator|(pipeable_config_element && lhs,
+                                    pipeable_config_element<other_derived_t, other_value_t> const & rhs)
+    {
+        return configuration{static_cast<derived_t &&>(lhs)}.push_back(static_cast<other_derived_t const &>(rhs));
+    }
+
+    //!\overload
+    template <typename other_derived_t, typename other_value_t>
+    friend constexpr auto operator|(pipeable_config_element const & lhs,
+                                    pipeable_config_element<other_derived_t, other_value_t> && rhs)
+    {
+        return configuration{static_cast<derived_t const &>(lhs)}.push_back(static_cast<other_derived_t &&>(rhs));
+    }
+
+    //!\overload
+    template <typename other_derived_t, typename other_value_t>
+    friend constexpr auto operator|(pipeable_config_element const & lhs,
+                                    pipeable_config_element<other_derived_t, other_value_t> const & rhs)
+    {
+        return configuration{static_cast<derived_t const &>(lhs)}.push_back(static_cast<other_derived_t const &>(rhs));
+    }
+
+    /*!\brief Combines two configuration elements to a seqan3::configuration.
+     * \tparam configs_t  A template parameter pack for the given seqan3::configuration.
+     * \param[in] lhs     The left hand operand.
+     * \param[in] rhs     The right hand operand.
+     * \returns A new seqan3::configuration adding `rhs` to the passed `lhs` object.
+     *
+     * \details
+     *
+     * Effectively calls seqan3::configuration::push_back on the left operand.
+     */
+    template <typename ...configs_t>
+    friend constexpr auto operator|(configuration<configs_t...> && lhs,
+                                    pipeable_config_element && rhs)
+    {
+        return std::move(lhs).push_back(static_cast<derived_t &&>(rhs));
+    }
+
+    //!\overload
+    template <typename ...configs_t>
+    friend constexpr auto operator|(configuration<configs_t...> const & lhs,
+                                    pipeable_config_element && rhs)
+    {
+        return lhs.push_back(static_cast<derived_t &&>(rhs));
+    }
+
+    //!\overload
+    template <typename ...configs_t>
+    friend constexpr auto operator|(configuration<configs_t...> && lhs,
+                                    pipeable_config_element const & rhs)
+    {
+        return std::move(lhs).push_back(static_cast<derived_t const &>(rhs));
+    }
+
+    //!\overload
+    template <typename ...configs_t>
+    friend constexpr auto operator|(configuration<configs_t...> const & lhs,
```

Comment:
these are in fact functions that would usually be perceived as being part of the configuration's interface, not the config element's interface, or not?